### PR TITLE
Add WatchOS support

### PR DESCRIPTION
**Pull request by FabioCornelli**
Hi, we are using moko-resources in our KMP mobile app (iOS & Android) mostly for strings localization and is working great!
Recently, we were asked to try to extend part of our application on watchOS and we found out that moko-resources does not target watchOS (see #74 )

So with these PR I've tried to add watchOS targets to moko-resources project.

Most of the code of "resources" project was already written in "appleMain" source set, so it was already compatible also on watchOS.
The "resources-generator" plugin code was also already generating .strings file correctly for watchOS targets.

So in this PR I've basically done the following:

- reworked "multiplatform-library-convention.gradle.kts" and "apple-main-convention.gradle.kts" in order to define watchOS source sets
- added watchOS targets in "resources" and "resources-test" projects
- implemented the expected classes and methods for watchOS targets
- added a watchOS target to the "kotlin-ios-app" sample project and test my modifications

I've run the sample on watchOS simulator and it seems to work correctly.

There a couple of things that i think can be improved:

1) I think the method `internal expect fun Color.toUIColor(): UIColor` should probably be moved in "moko-graphics" library. Shall I open a PR on moko-graphics ? if so, it's probably better to have a version of "moko-graphics" that support watchOS before moving on this PR.
2) I'm not sure that  having "kotlin-ios-app" with two target (iOS app and watchOS app) is a good approach or not, since it was giving me some issues in building the project correctly (which I solved by disabling gradle cache, see the last commit on this PR). 
If somebody has a better idea on how watchOS target can be tested, I'm open for suggestions and feedback.

Finally, with this PR I'm targeting #575 branch instead of develop because I've read in #624 that in #575 the plugin was rewritten.

Please let me know if that is correct or i should point to a different branch.

### CODE REVIEW

**Comment by ExNDY** [comment]
@FabioCornelli Hi, can u check latest commits on watchOs? I'm merge latest version of branch, change FontResource with latest version of Apple variant FontResource. Thank you

File: resources-build-logic/src/main/kotlin/apple-main-convention.gradle.kts [hunk 1]
```diff
@@ -16,6 +16,15 @@ kotlin {
...
         }
+        val watchosMain by getting {
+            dependsOn(appleMain)
+        }
+        val watchos64Main by getting {
+            dependsOn(watchosMain)
+        }
+        val watchos32Main by getting {
+            dependsOn(watchosMain)
+        }
```
Comment (Alex009) [inline resources-build-logic/src/main/kotlin/apple-main-convention.gradle.kts:27]: this part will be configured in resources-build-logic/src/main/kotlin/multiplatform-library-convention.gradle.kts

File: resources-test/build.gradle.kts [hunk 2]
```diff
@@ -11,6 +11,12 @@ plugins {
     id("publication-convention")
 }
 
+kotlin {
+    watchosX64()
+    watchosArm64()
+    watchosSimulatorArm64()
```
Comment (Alex009) [inline resources-test/build.gradle.kts:17]: just enable this targets in resources-build-logic/src/main/kotlin/multiplatform-library-convention.gradle.kts

File: resources/build.gradle.kts [hunk 3]
```diff
@@ -13,6 +13,10 @@ plugins {
 }
 
 kotlin {
+    watchosX64()
+    watchosArm64()
+    watchosSimulatorArm64()
```
Comment (Alex009) [inline resources/build.gradle.kts:18]: just enable this targets in resources-build-logic/src/main/kotlin/multiplatform-library-convention.gradle.kts

File: samples/kotlin-ios-app/gradle.properties [hunk 4]
```diff
@@ -1,7 +1,7 @@
 org.gradle.jvmargs=-Xmx12g
 org.gradle.parallel=true
 org.gradle.caching=true
-org.gradle.configuration-cache=true
+org.gradle.configuration-cache=false
```
Comment (Alex009) [inline samples/kotlin-ios-app/gradle.properties:4]: we should enable it

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: resources-build-logic/src/main/kotlin/apple-main-convention.gradle.kts (hunk 1 above)
Comment (ExNDY) [reply resources-build-logic/src/main/kotlin/apple-main-convention.gradle.kts]: convention plugins was reworked

File: resources-test/build.gradle.kts (hunk 2 above)
Comment (ExNDY) [reply resources-test/build.gradle.kts]: convention plugins was reworked

File: resources/build.gradle.kts (hunk 3 above)
Comment (ExNDY) [reply resources/build.gradle.kts]: convention plugins was reworked

File: samples/kotlin-ios-app/gradle.properties (hunk 4 above)
Comment (ExNDY) [reply samples/kotlin-ios-app/gradle.properties]: gradle.configuration-cache broken for watchos tasks in kotlin 1.9.10, but after upgrade kotlin version probles was fixed, enabled now

**Review by Alex009 — CHANGES_REQUESTED** [review]
conflict should be resolved

**Review by Alex009 — APPROVED** [review]

**Pull request merged**